File: solutions/go/darts/1/darts.go
package darts

func Score(x, y float64) int {
	var outer = 10.0
	var middle = 5.0
	var inner = 1.0
	var score = 0
	var distance = x*x + y*y

	if distance <= inner*inner {
		score = 10
	} else if distance <= middle*middle {
		score = 5
	} else if distance <= outer*outer {
		score = 1
	}
	return score
}

File: solutions/c/darts/1/darts.c
#include "darts.h"
#include <math.h>

uint8_t score(coordinate_t landing_position) {
    float distance = sqrt(landing_position.x * landing_position.x + landing_position.y * landing_position.y);
    if (distance <= 1) {
        return 10;
    } else if (distance <= 5) {
        return 5;
    } else if (distance <= 10) {
        return 1;
    }
    return 0;
}
